### ISSUE TITLE
VideoCapture Set call using CaptureProperty.Fps has no effect on versions 3.4.4 and 4.0.0

### ISSUE THREAD

**Issue by ravazquez** [shimat/opencvsharp#615]
# Summary of your issue

VideoCapture Set call using CaptureProperty.Fps has no effect on versions 3.4.4 and 4.0.0 but does work on 3.4.1.20181108.

Note: same seems to be true for CaptureProperty.Settings

# Environment

Windows 10 64 bit, Visual Studio 2017, Logitech HD 1080p webcam.

# What did you do when you faced the problem?

I was using the wrapper on previous versions and it worked as expected, after upgrading to versions 3.4.4 and 4.0.0 the code that used to be working does not behave in the same way.

## Example code:

    class VideoCaptureSample : ISample
    {
        public void Run()
        {
            var capture = new VideoCapture(CaptureDevice.DShow, -1);
            capture.Open(0);
            capture.Set(CaptureProperty.FrameHeight, 480.0);
            capture.Set(CaptureProperty.FrameWidth, 640.0);
            capture.Set(CaptureProperty.Fps, 30.0);
            int i = 0;
            using (var window = new Window("capture"))
            {
                Mat image = new Mat();
                while (true)
                {
                    if (capture.IsOpened())
                    {
                        capture.Read(image);
                        window.ShowImage(image);
                        if (i == 120)
                        {
                            capture.Open(0);
                            capture.Set(CaptureProperty.FrameHeight, 480.0);
                            capture.Set(CaptureProperty.FrameWidth, 640.0);
                            capture.Set(CaptureProperty.Fps, 1.0);
                        }
                    }
                    i++;
                    Cv2.WaitKey(1);
                } 
            }
        }
    }

## Output:

FPS drops significantly when var i hits 120 using version 3.4.1.20181108 but remains 30fps on versions 3.4.4 and 4.0.0.

## What did you intend to be?

FPS should still drop significantly when var i hits 120 using versions  3.4.4 and 4.0.0.

**Comment by died** [comment]
Why you repeat calling capture.Open ?

**Comment by died** [comment]
try this code (full code please check [github](https://github.com/died/OpenCv4CameraWithFps/blob/master/OpenCv4CameraTest/Program.cs) ).
```
            var cap = new VideoCapture(CaptureDevice.Any);
            var sw = Stopwatch.StartNew();
            var counter = 0;
            using (Window window = new Window("capture"))
            using (Mat image = new Mat())
            {
                while (true)
                {
                    cap.Read(image);
                    if (image.Empty())
                        break;
                    counter++;
                    double frame = (double) counter / sw.ElapsedMilliseconds * 1000;
                    Cv2.PutText(image, $"FPS:{frame:F}", new Point(10,40), HersheyFonts.HersheySimplex, 1, Scalar.Red);
                    window.ShowImage(image);
                    Cv2.WaitKey(1);

                    if (counter % 1299721 == 0) //random pick 100001st prime
                    {
                        counter = 0;
                        sw.Reset();
                    }
                }
            }
            cap.Release();
```

my webcam only can reach 30fps...
![imgur](https://i.imgur.com/wR3Oppf.png)

Actually I don't think opencv set fps can work well.

**Comment by died** [comment]
@ravazquez after some test, I know how to do it. 
If you want to set/get camera's fps, just using
```
cap.Fps = 120;
```
It can work on camera and only work on camera.

**Comment by blaisexen** [comment]
@ravazquez 
@died 

hi,
you can try my application that uses video camera, here
https://sourceforge.net/projects/opencvsharp-live-face-detect/

I haven't applied a CUDA, here's my question also,,
https://github.com/shimat/opencvsharp/issues/626

LoL,
good luck :dancing_women:

**Comment by ravazquez** [comment]
> (

@died and @blaisexen thanks for replying, my issue like I described is that i cannot change the FPS used by my webcam on versions 3.4.4 and 4.0.0, but I **can** change it in version 3.4.1.  Something changed on the wrapper or the core library between versions 3.4.1 and 3.4.4 that broke this behaviour, that is what I am tyring to report here.

**Comment by blaisexen** [comment]
@ravazquez 

are you sure there's no effect? 
because when I insert a few of your code with FPS from my source it taste good.

Now with **Cuda**....
https://sourceforge.net/projects/opencvsharp-live-face-detect/

will good luck

**Comment by ignacioinnovo** [comment]
Hi! This is a really great library! Thanks for make it open source!

I can´t set the FPS as @ravazquez described!

Same thing happen with the resolution.

I´m using a Genius FaceCam F100.

Thank you in advanced,
Ignacio

**Comment by ravazquez** [comment]
This issue seems fixed in 4.0.1.  Could have also been associated with setting a compatible FourCC for the desired FPS which I seem to have missed in my test. Some FPS settings are only available for specific FourCCs.
